src/ReactLeafletMap/layers/WMSLayer/index.js

```diff
@@ -0,0 +1,51 @@
+import { TileLayer } from 'leaflet'
+import isEqual from 'fast-deep-equal'
+
+import {GridLayer,withLeaflet } from 'react-leaflet';
+
+export const EVENTS_RE = /^on(.+)$/i
+class WMSTileLayer extends GridLayer {
+  createLeafletElement(props) {
+    const { url, ...params } = props
+    const { leaflet: _l, ...options } = this.getOptions(params)
+    return new TileLayer.WMS(url, options)
+  }
+
+  updateLeafletElement(fromProps, toProps) {
+    super.updateLeafletElement(fromProps, toProps)
+
+    const { url: prevUrl, opacity: _po, zIndex: _pz, ...prevProps } = fromProps
+    const { leaflet: _pl, ...prevParams } = this.getOptions(prevProps)
+    const { url, opacity: _o, zIndex: _z, ...props } = toProps
+    const { leaflet: _l, ...params } = this.getOptions(props)
+
+    if (url !== prevUrl) {
+      this.leafletElement.setUrl(url)
+    }
+    if (!isEqual(params, prevParams)) {
+
+        const forbidenParams = ['pane', 'minZoom', 'maxZoom', 'crs'];
```

Comment:
If we're modifying the component, maybe we should just keep valid params in one prop?

Reply:
There is not final enumeration of valid params. Each mapserver could have custom. It is reason, why I did forbidenParams. It is only this params added by reactLeaflet layer. If so, we could pass forbidden params instead of valid.

Reply:
Valid params, not param keys. We have those in ReactLeafletMap#getWmsTileLayer(), where we pass them here as individual props, which get mixed up with context, so we filter them.
Instead, we could pass them as one prop and be done with it. No need for any filtering, positive or negative.